### ISSUE TITLE
Ext.grid.column.ComponentColumn not removing old components' cache

### ISSUE THREAD

**Issue by fabriciomurta** [extnet/Ext.NET#1307]
The `componentColumn` is letting the cache have multiple copies of former/destroyed components.

Reported on this forum thread: [Control validation in ComponentColumn with TreePanel](http://forums.ext.net/showthread.php?61053)

The Ext.grid.column.ComponentColumn is an Ext.NET specific control so it has no representation on Sencha documentation.

A remediative override could be this:
```javascript
        Ext.define('Ext.grid.column.ComponentColumn', {
            override: 'Ext.grid.column.ComponentColumn',
            insertComponentForRecord: function (record, node, refreshSize) {
                var me = this,
                    cachedRecords = [],
                    i, j, entry, isDup;

                this.callParent(arguments);

                for (i in this.cache) {
                    entry = this.cache[i];

                    isDup = false;
                    for (j in cachedRecords) {
                        if (entry.id == cachedRecords[j]) {
                            isDup = true;
                            this.removeComponent(null, { id: entry.id });
                        }
                    }

                    if (!isDup) {
                        cachedRecords.push(entry.id); // adds to the list of already present
                    }
                }
            }
```

**Comment by fabriciomurta** [comment]
A solution worth considering would be adding the lines:
```javascript
// Before adding an item to the cache, remove a possible item that might be cached
this.removeComponent(null, record); // the view and rowId are not used anyway
```

on `Ext.grid.column.ComponentColumn.insertComponentForRecord()` right above the line that reads:
```javascript
this.cache.push({ id: record.id, cmp: cmp });
```

I.e. right before appending a component to the cache, remove any occurrence of it from the cache.

**Comment by DaniilVeriga** [comment]
Fixed after v3.3 and in v4.2.0.
